relayer/spy_relayer/src/listener/validation.ts

```diff
@@ -142,27 +170,22 @@ export async function parseAndValidateVaa(
     return "Token transfer is not for an approved token.";
   }
 
-  //TODO configurable
-  const sufficientFee = parsedPayload.fee && parsedPayload.fee > 0;
+  // TODO fee check
+  logger.debug("TODO check fee from swim payload");
 
-  if (!sufficientFee) {
-    logger.debug("Token transfer does not have a sufficient fee.");
-    return "Token transfer does not have a sufficient fee.";
-  }
-
-  const key = getKey(parsedPayload.originChain, originAddressNative as string); //was null checked above
+  const key = getKey(parsedVaaPayload.originChain, originAddressNative as string); //was null checked above
 
-  const isQueued = await checkQueue(key);
+  const isQueued = await exports.checkQueue(key);
   if (isQueued) {
     return isQueued;
   }
   //TODO maybe an is redeemed check?
 
-  const fullyTyped = { ...parsedVaa, payload: parsedPayload };
+  const fullyTyped = { ...parsedVaa, payload: { ...parsedVaaPayload, extraPayload: swimPayload }};
   return fullyTyped;
 }
 
-async function checkQueue(key: string): Promise<string | null> {
+export async function checkQueue(key: string): Promise<string | null> {
   try {
     const backupQueue = getBackupQueue();
     const queuedRecord = backupQueue.find((record) => {
```

Comment:
this is a bug. you need to either get rid of the `{}` or explicitly `return record[0] === key`. i fixed this on the main wormhole repo.

https://github.com/wormhole-foundation/wormhole/blob/dev.v2/relayer/spy_relayer/src/helpers/redisHelper.ts#L389